src/simulation/LoadGenerator.cpp

```diff
@@ -40,156 +40,78 @@ namespace stellar
 {
 
 using namespace std;
-
-// Account amounts are expressed in ten-millionths (10^-7).
-static const uint64_t TENMILLION = 10000000;
-
-// Every loadgen account or trustline gets a 999 unit balance (10^3 - 1).
-static const uint64_t LOADGEN_ACCOUNT_BALANCE = 999 * TENMILLION;
-
-// Trustlines are limited to 1000x the balance.
-static const uint64_t LOADGEN_TRUSTLINE_LIMIT = 1000 * LOADGEN_ACCOUNT_BALANCE;
+using namespace txtest;
 
 // Units of load are is scheduled at 100ms intervals.
 const uint32_t LoadGenerator::STEP_MSECS = 100;
 
-LoadGenerator::LoadGenerator(Hash const& networkID)
-    : mMinBalance(0), mLastSecond(0)
+LoadGenerator::LoadGenerator(Application& app)
+    : mMinBalance(0), mLastSecond(0), mApp(app)
 {
-    // Root account gets enough XLM to create 10 million (10^7) accounts, which
-    // thereby uses up 7 + 3 + 7 = 17 decimal digits. Luckily we have 2^63 =
-    // 9.2*10^18, so there's room even in 62bits to do this.
-    auto root = make_shared<AccountInfo>(0, txtest::getRoot(networkID),
-                                         10000000ULL * LOADGEN_ACCOUNT_BALANCE,
-                                         0, 0, *this);
-    mAccounts.push_back(root);
+    getRootAccount();
 }
 
 LoadGenerator::~LoadGenerator()
 {
     clear();
 }
 
-std::string
-LoadGenerator::pickRandomAsset()
+bool
+LoadGenerator::getRootAccount()
```

Comment:
I think that name and return value of this method are misleading.

This method does not return root account, and its return value is never used.
I propose to either:
* rename it to void LoadGenerator::createRootAccount
* make it return `res` or `nullptr`, remove `mRoot` field and pass returned value to `creationTransaction` as a parameter